drivers/storage/ide/uniata/id_ata.cpp

```diff
@@ -9499,6 +9499,82 @@ AtapiStartIo__(
                    (Srb->Cdb[0] != SCSIOP_ATA_PASSTHROUGH)/* &&
                    (Srb->Cdb[0] != SCSIOP_REPORT_LUNS)*/) {
                     KdPrint3((PRINT_PREFIX "Try ATAPI send %x\n", Srb->Cdb[0]));
+#ifdef __REACTOS__
+                    status = SRB_STATUS_BUSY;
+
+                    if (Srb->Cdb[0] == SCSIOP_INQUIRY &&
+                        (LunExt->DeviceFlags & DFLAGS_ATAPI_DEVICE) &&
+                        (LunExt->IdentifyData.DeviceType == ATAPI_TYPE_CDROM ||
+                        LunExt->IdentifyData.DeviceType == ATAPI_TYPE_OPTICAL) &&
+                        LunExt->IdentifyData.ModelNumber[0])
+                    {
+                        ULONG j;
+                        CCHAR vendorId[26];
+
+                        // Attempt to identify known broken CD/DVD drives
+                        for (j = 0; j < sizeof(vendorId); j += 2)
+                        {
+                            // Build a buffer based on the identify data.
+                            MOV_DW_SWP(vendorId[j], ((PUCHAR)LunExt->IdentifyData.ModelNumber)[j]);
+                        }
+
+                        // Emulate INQUIRY support for broken CD/DVD drives (e.g. Microsoft Xbox).
+                        // Currently we implement it by explicitly checking the drive name from ATA IDENTIFY PACKET.
+                        if (!AtapiStringCmp(vendorId, "THOMSON-DVD", 11) ||
+                            !AtapiStringCmp(vendorId, "PHILIPS XBOX DVD DRIVE", 22) ||
+                            !AtapiStringCmp(vendorId, "PHILIPS J5 3235C", 16) ||
+                            !AtapiStringCmp(vendorId, "SAMSUNG DVD-ROM SDG-605B", 24))
+                        {
+                            // TODO:
+                            // Better send INQUIRY and then check for chan->ReturningMediaStatus >> 4 == SCSI_SENSE_ILLEGAL_REQUEST
+                            // in AtapiInterrupt__() and emulate the response only in this case.
+
+                            // If this hack stays for long enough, consider adding Xbox 360 drive names to the condition,
+                            // as they are affected by the same problem.
```

Comment:
I don't get the point about not adding controller known to be flaky to the list.

Reply:
Right, but I don't know their exact names. OG Xbox DVD drives on the other hand are very well documented: https://xboxdevwiki.net/Porting_an_Operating_System_to_the_Xbox_HOWTO#DVD_Drive_Bugs